tests/foreman/api/test_remoteexecution.py

```diff
@@ -0,0 +1,105 @@
+"""Test for Remote Execution
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from nailgun import client
+from nailgun import entities
+from nailgun.entity_mixins import TaskFailedError
+from robottelo.api.utils import wait_for_tasks
+from robottelo.config import settings
+from robottelo.decorators import tier3, tier4
+from robottelo.helpers import add_remote_execution_ssh_key
+from robottelo.vm_capsule import CapsuleVirtualMachine
+
+
+@tier4
+def test_positive_run_capsule_upgrade_playbook():
+    """Run Capsule Upgrade playbook against an External Capsule
+
+    :id: 9ec6903d-2bb7-46a5-8002-afc74f06d83b
+
+    :steps:
+        1. Create a Capsule VM, add REX key.
+        2. Run the Capsule Upgrade Playbook.
+
+    :expectedresults: Capsule is upgraded successfully
+
+    :CaseImportance: Medium
+    """
+    with CapsuleVirtualMachine() as capsule_vm:
+        add_remote_execution_ssh_key(capsule_vm.ip_addr)
+        job = entities.JobInvocation().run(
+            synchronous=False,
+            data={
+                'job_template_id': 170,
+                'inputs': {
+                    'target_version': "6.8.z",
+                    'whitelist_options': "repositories-validate,repositories-setup",
+                },
+                'targeting_type': "static_query",
+                'search_query': f"name = {capsule_vm.hostname}",
+            },
+        )
+        wait_for_tasks(f"resource_type = JobInvocation and resource_id = {job['id']}")
+        result = entities.JobInvocation(id=job['id']).read()
+        assert result.succeeded == 1
+
+
+@tier3
+def test_negative_run_capsule_upgrade_playbook_on_satellite(default_org):
+    """Run Capsule Upgrade playbook against the Satellite itself
+
+    :id: 99462a11-5133-415d-ba64-4354da539a34
+
+    :steps:
+        1. Add REX key to the Satellite server.
+        2. Run the Capsule Upgrade Playbook.
+        3. Check the job output for proper failure reason.
+
+    :expectedresults: Should fail
```

Comment:
Isn't this a bit risky? What if it doesn't fail and initiates an upgrade on the system that would be potentially interfering with other tests? I suggest moving to destructive tier just in case

Reply:
Yes, you are right. If the check for satellite package failed for some reason (which this test is supposed to verify and I should add fail at the end of the try block), we would effectively run the upgrade and made the Satellite unresponsive for that time.